lib/constructorio.js

```diff
@@ -278,7 +333,53 @@ ConstructorIO.prototype = {
   },
 
   /**
-   * @description Adds a synonym group.
+   * @description Retrieves autocomplete results for supplied query
+   */
+  getAutocompleteResults(params, callback) {
+    const json = clonedeep(params);
```

Comment:
let's call this `qsParams` to be consistent with the other functions

Reply:
> Good job man, this looks awesome. Left a few comments.
> 
> I like how we're doing the checks for personalization parameters on the front end side.
> 
> One thing to note is that it seems like you can pass in any query parameter to autocomplete requests and there won't be any issues:
> 
> https://betaac.cnstrc.com/autocomplete/hel?autocomplete_key=key_aXLmVpkVp4BX21Sw&c=ciojs-1.41.0&num_results_Products=6&num_results_Search+Suggestions=8&i=98c05337-cab3-4141-a4bf-684374ef7ebb&s=309&query=hel&_dt=1565316426506&does_this_work=yup
> 
> But it doesn't work for search requests:
> 
> https://betaac.cnstrc.com/search/ham?c=ciojs-search-1.49.0&key=key_aXLmVpkVp4BX21Sw&i=98c05337-cab3-4141-a4bf-684374ef7ebb&s=309&page=1&num_results_per_page=24&section=Products&does_this_work_too=nope

Yes this is a bit annoying - because we want to always be passing the `c` parameter but certain routes won't permit it 😢